lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
         self.confirmation_token = nil
         generate_remember_token
+        save(validate: false)
```

Comment:
> update_columns will work the same way, but without touching other attributes (including timestamps) or going through validations or callbacks.

I'm not sure I can safely assume they don't want callbacks run or timestamps updated.